lib/safety_cone/filter.rb

```diff
@@ -12,12 +12,14 @@ def self.included(base)
     # based on the configuration.
     def safety_cone_filter
       if cone = fetch_cone
-        flash.clear
-        flash[notice_type(cone.measure)] = cone.message
+        flash[:safetycone_notice] = flash[:safetycone_alert] = nil
+        flash["safetycone_#{notice_type(cone.measure)}"] = cone.message
         redirect_to safety_redirect if cone.measure == 'block'
```

Comment:
Have a think about the name 'measure' as it is misleading. I would think something like type would be good.